tests/integration/cloud/end_to_end/conftest.py

```diff
@@ -57,54 +56,37 @@ def datasource_name(
         _ = context.data_sources.add_pandas(name=datasource_name)
         context.data_sources.get(name=datasource_name)
     context.delete_datasource(name=datasource_name)
-    with pytest.raises(ValueError):
+    with pytest.raises(KeyError):
         _ = context.data_sources.get(name=datasource_name)
 
 
 @pytest.fixture(scope="module")
 def expectation_suite(
     context: CloudDataContext,
 ) -> Iterator[ExpectationSuite]:
+    """This ExpectationSuite is shared by each E2E test, so its expected that the data
+    used by each test follows the same shape."""
     expectation_suite_name = f"es_{uuid.uuid4().hex}"
     expectation_suite = context.suites.add(ExpectationSuite(name=expectation_suite_name))
     assert len(expectation_suite.expectations) == 0
-    yield expectation_suite
-    expectation_suite = context.suites.get(name=expectation_suite_name)
-    assert len(expectation_suite.expectations) > 0
+    expectation_suite.add_expectation(ExpectColumnValuesToNotBeNull(column="name", mostly=1))  # type: ignore[arg-type]  # todo: fix in core-412
+    assert len(expectation_suite.expectations) == 1
```

Comment:
nit: while you're here, I'd probably delete these asserts. That functionality should be covered in unit tests. I can see an argument for the `with pytest.raises` staying, though, since that is more of an end-to-end check.

Reply:
great suggestion, will update 👍